src/fondant/pipeline/compiler.py

```diff
@@ -487,3 +487,133 @@ def _set_configuration(task, fondant_component_operation):
+            pipeline: the pipeline to compile
+            output_path: the path where to save the Kubeflow pipeline spec.
+        """
+        run_id = pipeline.get_run_id()
+        path = pipeline.base_path
+        pipeline.validate(run_id=run_id)
+
+        component_cache_key = None
+
+        steps: t.List[t.Any] = []
+        for component_name, component in pipeline._graph.items():
+            component_op = component["fondant_component_op"]
+
+            component_cache_key = component_op.get_component_cache_key(
+                previous_component_cache=component_cache_key,
+            )
+
+            metadata = Metadata(
+                pipeline_name=pipeline.name,
+                run_id=run_id,
+                base_path=path,
+                component_id=component_name,
+                cache_key=component_cache_key,
+            )
+
+            logger.info(f"Compiling service for {component_name}")
+
+            # add metadata argument to command
+            command = ["--metadata", metadata.to_json()]
+
+            # add in and out manifest paths to command
+            command.extend(
+                [
+                    "--output_manifest_path",
+                    f"{path}/{metadata.pipeline_name}/{metadata.run_id}/"
+                    f"{component_name}/manifest.json",
+                ],
+            )
+
+            # add arguments if any to command
+            for key, value in component_op.arguments.items():
+                if isinstance(value, (dict, list)):
+                    command.extend([f"--{key}", json.dumps(value)])
+                else:
+                    command.extend([f"--{key}", f"{value}"])
+
+            # resolve dependencies
+            if component["dependencies"]:
+                for dependency in component["dependencies"]:
+                    # there is only an input manifest if the component has dependencies
+                    command.extend(
+                        [
+                            "--input_manifest_path",
+                            f"{path}/{metadata.pipeline_name}/{metadata.run_id}/"
+                            f"{dependency}/manifest.json",
+                        ],
+                    )
+
+                    depends_on = [steps[-1]]
+            else:
+                depends_on = []
+
+            script = self.generate_component_script(component_name, command)
+
+            processor = self.sagemaker.processing.ScriptProcessor(
+                image_uri=component_op.component_spec.image,
+                command=["bash"],
+                instance_type="ml.t3.medium",
+                instance_count=1,
+                base_job_name=component_name,
+                role=self.sagemaker.get_execution_role(),
+            )
+            step = self.sagemaker.workflow.steps.ProcessingStep(
+                name=component_name,
+                processor=processor,
+                depends_on=depends_on,
+                code=script,
```

Comment:
Just to better understand. Can you explain again how you resolved the issue of the long command? is it simply by passing the command to execute to `code` which has no limit? 

The code here is then the command to execute and the `command` in the `processor` is the executing command?

Reply:
The script is a an actual file we create per component, it contains the execute command and all arguments (which were the main offenders for the length limit).

I have included the script generated by this in the example pipeline in this PR (see `playground/some-cool-pipeline/artifacts/`)

we then point the ProcessingStep to this file and SageMaker will upload it to S3 and set it as input automatically you can see this in the resulting `playground/some-cool-pipeline/spec.json` 

This avoids the length issues but requires the right credentials during compile to upload these files

Reply:
Thanks all clear now! Good that we don't need to add any custom logic at the executor